server/src/completions.ts

```diff
@@ -23,6 +23,55 @@ function toCompletionItems(
   return allowedTypes.map((label) => ({ label, kind }))
 }
 
+export function isInsideAttribute(
+  currentLineUntrimmed: string,
+  position: Position,
+): boolean {
+  let numberOfOpenBrackets = 0
+  let numberOfClosedBrackets = 0
+  for (let i = 0; i < position.character; i++) {
+    if (currentLineUntrimmed[i] === '(') {
+      numberOfOpenBrackets++
+    } else if (currentLineUntrimmed[i] === ')') {
+      numberOfClosedBrackets++
+    }
+  }
+  return numberOfOpenBrackets > numberOfClosedBrackets
+}
+
+function getSymbolBeforePosition(
+  document: TextDocument,
+  position: Position,
+): string {
+  return document.getText({
+    start: {
+      line: position.line,
+      character: position.character - 1,
+    },
+    end: { line: position.line, character: position.character },
+  })
+}
+
+export function positionIsAfterFieldAndType(
+  currentLine: string,
+  position: Position,
+  document: TextDocument,
+): boolean {
+  const wordsBeforePosition: string[] = currentLine
+    .slice(0, position.character - 1)
+    .trim()
+    .split(/\s+/)
+
+  const symbolBeforePosition = getSymbolBeforePosition(document, position)
+  const symbolBeforeIsWhiteSpace = symbolBeforePosition.search(/\s/)
+
+  return (
+    wordsBeforePosition.length > 2 ||
+    (wordsBeforePosition.length === 2 &&
+      (symbolBeforeIsWhiteSpace !== -1 || symbolBeforePosition === '@'))
```

Comment:
Up to you, but for longer conditionals like this, it might make sense to name them with a variable to explain what they're doing. Something like:

```js
const hasAtRelation = wordsBeforePosition.length === 2 && (symbolBeforeIsWhiteSpace !== -1 || symbolBeforePosition === '@')
```